unlock-app/src/components/interface/checkout/main/Minting.tsx

```diff
@@ -185,16 +190,32 @@ export function Minting({
   const { account } = useAuth()
   const [state, send] = useActor(checkoutService)
   const config = useConfig()
-  const { mint, lock, messageToSign, metadata } = state.context
+  const { mint, lock, messageToSign, metadata, recipients } = state.context
   const processing = mint?.status === 'PROCESSING'
 
   useEffect(() => {
     if (mint?.status !== 'PROCESSING') {
       return
     }
+
+    const waitForTokenIds = async (): Promise<string[]> => {
```

Comment:
In cross chain contexts, waiting for the tx to be mined on the sending chain is not enough. We need to wait for the tokens on the destination chain.